core/subscription/context.go

```diff
@@ -202,6 +187,46 @@ func (ctx *opCtx) detailsEvents() {
 	ctx.groupDetailsEvents(msgs)
 }
 
+func (ctx *opCtx) addDetailsEvents(prev, curr *entry, info struct {
+	id     string
+	subIds []string
+	keys   []string
+}, msgs []*pb.EventMessage) []*pb.EventMessage {
+	var sentAmendDetails, sentSetDetails []string
+	if prev != nil {
+		active := prev.GetActive()
+		detailsSent := prev.GetFullDetailsSent()
+		sentAmendDetails = lo.Intersect(active, detailsSent)
+
+		notActive := slice.Difference(info.subIds, active)
+		detailsNotSent := slice.Difference(info.subIds, detailsSent)
+		sentSetDetails = lo.Union(notActive, detailsNotSent)
```

Comment:
naming("sent") is confusing here, it is actually opposite.  Maybe smth like "subIdsToSendFullDetails", "subIdsToSendDiffDetails"